src/Esprima/Scanner.cs

```diff
@@ -1162,8 +1180,17 @@ private Token ScanBinaryLiteral(int start)
             }
         }
 
-        var numberString = number.ToString();
-        var value = Convert.ToUInt32(numberString, 2);
+        double value;
+
+        if (number.Length <= 64)
+        {
+            value = Convert.ToUInt64(number.ToString(), fromBase: 2);
```

Comment:
looking how this is implemented in NET 6.0, would be nice to get access to that `ParseNumbers.StringToLong(ReadOnlySpan<char> s, int radix, int flags)` which is internal...

Reply:
Yep, this situation if unfortunate. Maybe they'll expose this API in a later version.

Reply:
Actually looking at the current code, we could probably just copy-paste it and un-localize the error messages and we would have the solution. Not sure if it's worth it though.

https://github.com/dotnet/runtime/blob/6cd25fbeb1a2a332561a46cdf2b01d06c1fbb192/src/libraries/System.Private.CoreLib/src/System/ParseNumbers.cs

Reply:
Not sure either but it's very likely that there's further room for improvement in this area. Currently I don't plan to spend more time on it but feel free to experiment with this if you .